frontend/src/api/AuthenticationUtil.js

```diff
@@ -1,14 +1,44 @@
+import Cookies from "js-cookie";
+
+function parseJwt(token) {
+  const base64Url = token.split(".")[1];
+  const base64 = base64Url.replace(/-/g, "+").replace(/_/g, "/");
+  const jsonPayload = decodeURIComponent(
+    atob(base64)
+      .split("")
+      .map((c) => {
+        const component = `00${c.charCodeAt(0).toString(16)}`;
+        return `%${component.slice(-2)}`;
+      })
+      .join("")
+  );
+
+  return JSON.parse(jsonPayload);
+}
+
 export function saveToken(token) {
-  localStorage.setItem("authToken", token);
+  const tokenPayload = parseJwt(token.split(" ")[1]);
+  const expiryDate = new Date(tokenPayload.exp * 1000);
+  Cookies.set("authToken", token, { expires: expiryDate, SameSite: "Strict" });
 }
 
 export function getAuthorizationHeader() {
-  const token = localStorage.getItem("authToken");
+  const token = Cookies.get("authToken");
   return {
     headers: { Authorization: `${token}` }
   };
 }
 
+export function isAuthenticated() {
+  return !!Cookies.get("authToken");
+}
+
+export function isAdmin() {
+  const token = Cookies.get("authToken");
+  const tokenPayload = parseJwt(token.split(" ")[1]);
+  return tokenPayload.sub === "admin";
+}
```

Comment:
This is incorrect. `sub` is the username. You'll need to merge #354 which added roles. You can then do something like tokenPayload.role === "ADMIN"

Reply:
I'll fix this once #354 is merged

Reply:
Fixed in f14cc12